packages/semcom-node/lib/component/services/base-component.service.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  ComponentMetadata,
+  ManageComponentInMemoryService,
+  QueryComponentInMemoryService,
+} from '@digita-ai/semcom-core';
+import { ComponentService } from './component.service';
+
+export class BaseComponentService implements ComponentService {
+  private queryService: QueryComponentInMemoryService;
+  private manageService: ManageComponentInMemoryService;
```

Comment:
The goal of this pattern is to be able to inject new implementations of a service when necessary. Therefore, inject an instance of the abstract class instead of its implementation.